src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -194,6 +194,10 @@ void BootstrapRequest<I>::handle_get_client(int r) {
     derr << ": failed to retrieve client: " << cpp_strerror(r) << dendl;
     finish(r);
     return;
+  } else if (m_client.state == cls::journal::CLIENT_STATE_DISCONNECTED) {
+    dout(0) << ": client flagged disconnected, re-registering" << dendl;
+    unregister_client();
```

Comment:
Instead of unregister / re-register, why not just update the connected state back to connected (and have the cls method clear the current commit position). How about a new config value for whether or not to automatically resync after a disconnect? Until we get the delta image sync logic, I'd hate to have an overloaded system get behind on replay and then just automatically start a full resync.